spaces/views.py

```diff
@@ -87,4 +95,50 @@ def get(self, request, space_id):
             ]
             return JsonResponse({"detail_space" : room}, status = 200)
         except Space.DoesNotExist:
-            return JsonResponse({"message" : "SPACE_DOES_NOT_EXIST"}, status = 400)
\ No newline at end of file
+            return JsonResponse({"message" : "SPACE_DOES_NOT_EXIST"}, status = 400)
+
+
+class ReservationView(View):
+    @login_required
+    @transaction.atomic
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            len_ = 20
+            string_pool = string.ascii_letters + string.digits
+            reservation_code = ""
+            for i in range(len_):
+                reservation_code += random.choice(string_pool)
+
+            if Reservation.objects.filter(user=request.user,space=Space.objects.get(id=data['space']), check_in=data['check_in'], check_out=data['check_out']):
```

Comment:
`space=Space.objects.get(id=data['space'])`를 미리 선언해주는게 좋을 것 같습니다. 아래 `line 117`과 `line 122`에서 계속해서 `get()`으로 호출하기 때문에, db를 불필요하게 조회하고 있습니다.
수정해주세요~

```suggestion
            if Reservation.objects.filter(user=request.user, space=Space.objects.get(id=data['space']), check_in=data['check_in'], check_out=data['check_out']):
```

Reply:
헐 그러네요 수정하겠습니다!!